### ISSUE TITLE
Cannot Unsubscribe from Event from Handler

### ISSUE THREAD

**Issue by tf** [zohararad/audio5js#63]
Calling `Pubsub#off` is no-op while `Pubsub#trigger` is running:

```javascript
function handler() {
  if (...) {
    player.off('progress', handler)
  }
})

player.on('progress', handler)
```

This happens since `trigger` [empties the `channels` array](https://github.com/zohararad/audio5js/blob/master/src/audio5.js#L138) while processing event handlers.

In most cases you can probably just use `one` instead. Still, it's a gotcha.

**Comment by zohararad** [comment]
I agree that this is an edge-case that can be easily solved by using `one` instead of `on`. Do you have any suggestions for an elegant solution to this?

**Comment by tf** [comment]
Well, in the simplest case the iteration could simply be rewritten to a `for`-loop, right?

**Comment by zohararad** [comment]
Correct, but then removing the events marked as `once` becomes a bit of a hassle. I'm happy to accept a PR if you want to send one

**Comment by kamijoutouma** [comment]
@zohararad @tf from my experience with javascript i'm pretty sure that if you do one or once it will return the same value it returned the first time every time it see's the event 'progress'. If this is what you are aiming for then i'm all for it. But Otherwise if no response from the owner of this issue is made within 24-48 hours can this be closed?

**Comment by zohararad** [comment]
@kamijoutouma this is an edge-case to begin with. Agreed - we'll close of no response from owner in 24h

**Comment by tf** [comment]
I still think this is not the desired behavior. I don't have time to provide a fix myself right now though. So if you think this does not have to be addressed, feel free to close.

**Comment by zohararad** [comment]
@tf I agree it's not the desired behaviour, but given that I too have little time to maintain the library, and we're talking about an edge case, I'm going to leave it open for now and treat it as low priority.

I'll try and allocate some time to investigate a solution. @kamijoutouma please feel free to pitch in if you have the time.

**Comment by kamijoutouma** [comment]
@zohararad Sure ill be happy to.

**Comment by kosso** [comment]
I had issues unsubscribing using `off()`

The only way I got around it was to flush the play channels arrays for each event.

eg: `yourAudio5Player.channels.play = [];`  seems to remove the event listener for `play`.

**Comment by zohararad** [comment]
Can you share a code sample of this? It might be a persistent bug in the code I need to address.
